### DISCUSSION TITLE
Option to convert normal note to folder note when file attachments are added

### DISCUSSION THREAD

**Discussion by rbosh** [LostPaul/obsidian-folder-notes#229] [Ideas]
This plugin is exactly was I was looking for, this is wonderful, thank you!

It would be great to set the option for the plugin to automatically convert a normal note (not yet a folder note) to a folder note as soon as file attachments are added to the note.

However, this needs to be compatible with Obsidian's `Default location for new attachments`-setting.
I currently have set this to `In subfolder under current folder` (called `assets`), and I would expect the behavior that the file attachment gets saved into the folder note folder in a subfolder called `assets` (and not into a folder  `assets` on the same level as the folder note).
In other words, the plugin would need to convert the note to a folder note before Obsidian moves the file attachment to the specified location.

**Comment by boettges** [comment]
Having "Obsidian Folder Notes" automatically turn a regular note into a folder note as soon as an attachment gets added is the missing puzzle piece. 

It would partially solve this: https://forum.obsidian.md/t/new-default-location-behavior-for-attachments-images-automatically-convert-the-notes-md-into-a-container-folder-itself-that-holds-all-embeds-and-cleverly-hide-it-throughout-the-ui-by-treating-it-as-a-regular-note/ 

It would pose as an elegant solution to a problem that many users seem to face. Especially those who are used to traditional note taking apps with proprietary file formats where attachments get embedded into the file bundle.

**Comment by adepke** [comment]
I've implemented this feature here: https://github.com/LostPaul/obsidian-folder-notes/pull/314
